app/Models/Support.php

```diff
@@ -213,5 +213,76 @@ public static function promoteDelegatesToDirect($topicNum, $nickNames)
 
         return;
     }
-    
+
+    public static function getAllSupporters($topic,$camp,$excludeNickID){
+        $nickNametoExclude = [$excludeNickID];
+        $support = self::where('topic_num','=',$topic)->where('camp_num','=',$camp)
+                ->where('end','=',0)
+                ->where('nick_name_id','!=',$excludeNickID)
+                ->where('delegate_nick_name_id',0)->groupBy('nick_name_id')->get(); 
+        $camp = Camp::where('camp_num','=',$camp)->where('topic_num','=',$topic)->first();
+        $allChildren = Camp::getAllChildCamps($camp);
+        $supportCount = 0;
+        if(sizeof($support) > 0 || count($support) >0){
+            foreach($support as $sp){
+                array_push( $nickNametoExclude, $sp->nick_name_id);
```

Comment:
May be we can simplify this 
`$nickNametoExclude[] = $sp->nick_name_id;` Will it work?